src/fee.rs

```diff
@@ -71,22 +42,17 @@ impl Fee {
                     total_txs += block_extra.block.txdata.len() as u64;
 
                     if block_extra.height % 10_000 == 0 {
-                        let (utxo_size, collision_size, utxo_capacity) = self.utxo.map.len();
-                        info!(
-                            "(utxo, collision, capacity): {:?} load:{:.1}% script on stack: {:.1}% unspendable:{}",
-                            (utxo_size, collision_size, utxo_capacity),
-                            (utxo_size as f64 / utxo_capacity as f64) * 100.0,
-                            self.utxo.map.script_on_stack() * 100.0,
-                            self.utxo.unspendable,
-                        );
-                    }
-                    for tx in block_extra.block.txdata.iter() {
-                        self.utxo.add(tx);
+                        info!("{}", self.utxo.stat());
                     }
 
+                    self.utxo.add(&block_extra.block, block_extra.height);
+                    let mut prevouts = self.utxo.get(block_extra.height);
```

Comment:
Maybe I read it wrong, but `prevouts` was just there in the memory when doing `.add(.., block.height);` and instead of using it (returning it from `add`? or caching in memory in the `DbUtxo` somewhere, you go to through the database to read it from the database again?

Reply:
Or maybe `add` and `get` should just be made into one call.

Reply:
> Or maybe `add` and `get` should just be made into one call.

I like that, what do you think `add_outputs_get_inputs(block: &Block, height: u32) -> Vec<TxOut>` ?

Reply:
Some context... I started with something like `insert(output)` and `get(prevout)` which is just ok for the memory case, but the db case needed to "batch the API" to take advantage of multiple runs... I ended up "batching" also the memory impl to have more similar API,  but I didn't notice this last improvement of merging `add` and `get` at this point...